### PR TITLE
Update logging in all configs

### PR DESCRIPTION
**Pull request by krammnic**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (d here)

Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?
* #2648

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [X] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by krammnic** [comment]
cc @joecummings

File: recipes/full_finetune_distributed.py [hunk 1]
```diff
@@ -178,11 +176,12 @@ def __init__(self, cfg: DictConfig) -> None:
         self._log_every_n_steps = cfg.get("log_every_n_steps", 1)
         self._log_peak_memory_stats = cfg.get("log_peak_memory_stats", False)
         if self._log_peak_memory_stats and self._device.type not in {"cuda", "xpu"}:
-            log.info(
+            self._logger.info(
                 "log_peak_memory_stats was set to True, however, training does not use cuda or xpu."
                 "Setting log_peak_memory_stats=False."
             )
             self._log_peak_memory_stats = False
+        self._logger = utils.get_logger(cfg.log_level)
```
Comment (joecummings) [inline recipes/full_finetune_distributed.py:184]: Think it might fail if you initialize the logger AFTER you call it :)

**Review by joecummings — COMMENTED** [review]
1) Can you add the comment in the config to outline a couple of the options for logging levels (See the Github Issue)
2) Show me at least one or two sanity check runs

**Review by joecummings — APPROVED** [review]
🫡

**Comment by bhargavyagnik** [comment]
I had tried to solve this issue day before yesterday, but had found a more complex issue with logging and was 99% sure I was wrong. 

Today, I tried to use this merged code to see if that error still persists. And in fact it does.

I tried using the lora_finetune_single_device.py recipe with a config file which contains log_level: WARN.
The output is as follows:
``` INFO:torchtune.utils._logging:Running LoRAFinetuneRecipeSingleDevice with resolved config:
Writing logs to /tmp/torchtune/qwen2_0_5B/lora_single_device/logs/log_1746236088.txt
/home/euler/code/torchtune/torchtune/recipes/lora_finetune_single_device.py:461: FutureWarning: lora_attn_modules is deprecated for validate_missing_and_unexpected_for_lora and will be removed in future versions. Please use state_dict_keys instead.
  validate_missing_and_unexpected_for_lora(
/home/euler/code/torchtune/torchtune/torchtune/utils/_logging.py:143: FutureWarning: apply_lora_to_mlp is deprecated for validate_missing_and_unexpected_for_lora and will be removed in future versions. Please use state_dict_keys instead.
  return obj(*args, **kwargs)
/home/euler/code/torchtune/torchtune/torchtune/utils/_logging.py:143: FutureWarning: apply_lora_to_output is deprecated for validate_missing_and_unexpected_for_lora and will be removed in future versions. Please use state_dict_keys instead.
  return obj(*args, **kwargs)
INFO:torchtune.utils._logging:Model is initialized with precision torch.bfloat16.
INFO:torchtune.utils._logging:Memory stats after model init:
        GPU peak memory active: 1.30 GiB
        GPU peak memory alloc: 1.30 GiB
        GPU peak memory reserved: 1.36 GiB
INFO:torchtune.utils._logging:Tokenizer is initialized from file.
INFO:torchtune.utils._logging:Optimizer and loss are initialized.
INFO:torchtune.utils._logging:Loss is initialized.
INFO:torchtune.utils._logging:Learning rate scheduler is initialized.
WARNING:torchtune.utils._logging: Profiling disabled.
INFO:torchtune.utils._logging: Profiler config after instantiation: {'enabled': False} 
```


But it should not have INFO logs because of higher severity of WARN. So i debugged the code to find that logger severity level changes automatically after internal torchtune imports of one of the loss functions like in our case at 

https://github.com/pytorch/torchtune/blob/d39fd9b9b1a2ffb0fbd450d1a478e6128607dbd4/recipes/lora_finetune_single_device.py#L295

where the logger changes to DEBUG severity level because of the code in 
https://github.com/pytorch/torchtune/blob/d39fd9b9b1a2ffb0fbd450d1a478e6128607dbd4/torchtune/modules/loss/ce_chunked_output_loss.py#L15.

Proofs:

1. After logger is initialized 
Screenshot 2025-05-02 at 9 48 41 PM

2. After Line#295 (the loss function call)
Screenshot 2025-05-02 at 9 50 59 PM

Then it stays the same DEBUG level.

To verify that again, 

I added the following line
`self._logger.debug("Here debug should not be outputted.")`. after line 301 in the lora_finetune_single_device.py.


And kept the log_level: ERROR in custom_config.yaml file.
Here the output was as follows:
```
INFO:torchtune.utils._logging:Model is initialized with precision torch.bfloat16.
INFO:torchtune.utils._logging:Memory stats after model init:
        GPU peak memory active: 1.30 GiB
        GPU peak memory alloc: 1.30 GiB
        GPU peak memory reserved: 1.36 GiB
INFO:torchtune.utils._logging:Tokenizer is initialized from file.
INFO:torchtune.utils._logging:Optimizer and loss are initialized.
DEBUG:torchtune.utils._logging:Here debug should not be outputted.
INFO:torchtune.utils._logging:Loss is initialized.
INFO:torchtune.utils._logging:Learning rate scheduler is initialized.
WARNING:torchtune.utils._logging: Profiling disabled.
INFO:torchtune.utils._logging: Profiler config after instantiation: {'enabled': False}
1|2|Loss: 1.8641810417175293:   0%|                               | 2/1617 [01:08<15:16:03,
```

I hope this helps us to improve our project. If I am wrong, please excuse me. Thank you so much for the amazing repository.

**Comment by bhargavyagnik** [comment]
I think @krammnic 's contribution is quite to the point, if we can remove the package's 'DEBUG' code and fix that, i think it should work. I shall try that out and will update you.
